### DISCUSSION TITLE
V4 - ApolloServer TSError Expected 0 type arguments but got 1.

### DISCUSSION THREAD

**Discussion by DanielPachecoAr** [apollographql/apollo-server#7368] [Q&A]
I have the following code baesd on the migration examples from the [apollo page](https://www.apollographql.com/docs/apollo-server/migration/#migrate-from-apollo-server-express).

<code>
    import { ApolloServer } from '@apollo/server';
    import { expressMiddleware } from '@apollo/server/express4';

      interface MyCustomContext {
        dataSources: MyDataSources;
        token: any;
      }

      const apolloServer = new ApolloServer<MyCustomContext>({
        typeDefs: schema,
        resolvers,
        plugins: [...],
        apollo: {
          key: APOLLO_KEY,
          graphRef: APOLLO_REF
        }
      });

    await apolloServer.start();

      let server = expressMiddleware(apolloServer, {
          context: async ({ req }) => {
          let token = req.headers.token;
          return {
            dataSources: {
              .
              .
              .
            },
            token
          };
      }
      });

     return server;
</code>

From that code Im getting the following typescript compilation errors:

<code>
    TSError: ⨯ Unable to compile TypeScript:
    index.ts:XX:41 - error TS2558: Expected 0 type arguments, but got 1.

    XX   const apolloServer = new ApolloServer<MyCustomContext >({
                                                                   ~~~~~~~~~~~~~~~~
    index.ts:YY:22 - error TS2339: Property 'start' does not exist on type 'ApolloServer'.

    YY   await apolloServer.start();
</code>

Im able to bypass these errors using:

<code>
    // THIS
    // @ts-ignore 
      const apolloServer = new ApolloServer<MyCustomContext>({
        typeDefs: schema,
        resolvers,
        plugins: [...],
        apollo: {
          key: APOLLO_KEY,
          graphRef: APOLLO_REF
        }
      }) as any; // AND THIS
</code>


And the serverworks, however Im hoping get some help in order not to use those workarounds.

BTW [HERE](https://github.com/apollographql/apollo-server/blob/6ff88e87c52f4d6734c36fda96be07d4f8ad80cc/packages/server/src/ApolloServer.ts#L208) is the ApolloServer definition, and I get to that file when ctrl+clicking the call on my code.

**Comment by trevor-scheer** [comment]
Massive, syntactically incorrect, and incomplete snippets don't really help me to get you to the bottom of your problem. Can you please send a codesandbox link or provide an actual reproduction that I can clone or run locally which demonstrates the issue in a concise way?

**Reply by DanielPachecoAr** [reply]
Will do ASAP, thanks for the quick response.

**Reply by glasser** [reply]
I suspect the version of `typescript` might be relevant (though that will also be clear from a self-contained reproduction).

**Reply by DanielPachecoAr** [reply]
I couldn´t manage to reproduce it in codesandbox, and it got fixed with a node update.
